Refactor foreach pointwise ops tests to use OpInfo

izdeby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52450 wip
* #49039 [wip] Replace optimizers in torch.optim with the ones from torch.optim._multi_tensor
* #52449 [WIP] Add type promotion to a fast path for foreach APIs
* #52448 [WIP] Support broadcasting via slow path
* #48224 [wip] Pushed unsupported scenarios to slow path
* #51170 Updated foreach pointwise ops to support complex and bloat16 on CUDA
* #49714 Update foreach max/min
* **#51060 Refactor foreach pointwise ops tests to use OpInfo**
* #51059 Refactor foreach binary ops tests with tensor lists to use OpInfo
* #51058 Refactor foreach binary ops tests with scalars to use OpInfo
* #49712 Refactor foreach unary ops tests to use OpInfo

-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

Commit:
refactor foreach pointwise ops tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "refactor foreach pointwise ops tests to use OpInfo"




[ghstack-poisoned]

Commit:
Update on "refactor foreach pointwise ops tests to use OpInfo"




[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -1047,53 +1047,53 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad):
+                    method=torch._foreach_addcmul,
+                    inplace=torch._foreach_addcmul_,
+                    ref=torch.addcmul,
+                    dtypes=all_types(),
```
@ngimel had an interesting idea that these could acquire their dtypes from the reference operation's OpInfo, at least when that OpInfo exists. For example, the OpInfo for asin can be acquired by matching reference and the .op attribute of the OpInfo.

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

Commit:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]

izdeby committed:
Update on "Refactor foreach pointwise ops tests to use OpInfo"


-------
Refactor foreach pointwise ops tests to use OpInfo

Differential Revision: [D26103908](https://our.internmc.facebook.com/intern/diff/D26103908)

[ghstack-poisoned]
